arangod/RocksDBEngine/RocksDBIterators.cpp

```diff
@@ -87,7 +95,7 @@ bool RocksDBAllIndexIterator::nextImpl(LocalDocumentIdCallback const& cb, size_t
 
   TRI_ASSERT(limit > 0);
 
-  do {
```

Comment:
but limit > 0 for first iteration

Reply:
I think using `do { ... } while (limit > 0)` and `while (limit > 0) { ... }` are equivalent here, as before the loop we always check for a limit value of 0.
Anyway, @goedderz requested the `do { ... } while (limit > 0)` stuff to be turned into `while (limit > 0) { ... }` in some other places, so I did this here as well just to get this PR finally through.